### PR TITLE
📖 update releasing.md

### PR DESCRIPTION
**Pull request by tuminoid**
Update releasing.md with the up to date steps for creating CAPM3 release.

### CODE REVIEW

**Review by smoshiur1237 — APPROVED** [review]
It is in better shape and understandable step by step instruction. 
LGTM from my side, lets get review by others.

File: docs/releasing.md
Comment (kashifest) [inline docs/releasing.md:37]:
```suggestion
`git clone git@github.com:metal3-io/cluster-api-provider-metal3`
```

File: docs/releasing.md
Comment (kashifest) [inline docs/releasing.md:10]:
```suggestion
- Uplift controller go modules to use latest corresponding CAPI modules
```

File: docs/releasing.md
Comment (kashifest) [inline docs/releasing.md:25]:
```suggestion
Release team gets the permissions via `metal3-release-team` membership. Adding
```

File: docs/releasing.md
Comment (kashifest) [inline docs/releasing.md:27]: Individual persons should not be given the permissions directly.

File: docs/releasing.md
Comment (kashifest) [inline docs/releasing.md:39]:
```suggestion
or if using existing repository, verify your intended remote is set  to `metal3-io`
```

File: docs/releasing.md
Comment (kashifest) [inline docs/releasing.md:51]: main tag means ?

File: docs/releasing.md
Comment (kashifest) [inline docs/releasing.md:54]:
```suggestion
- Create a signed, annotated tag `git tag -s -a v1.x.y -m v1.x.y`
```

File: docs/releasing.md
Comment (kashifest) [inline docs/releasing.md:55]:
```suggestion
- Push the tags to the GitHub repository `git push origin v1.x.y`
```

File: docs/releasing.md
Comment (kashifest) [inline docs/releasing.md:68]:
```suggestion
  This enables the tag to be used as a Go module version for any downstream
```

File: docs/releasing.md
Comment (kashifest) [inline docs/releasing.md:79]:
```suggestion
  the PR ids and summary title. This way the changes are acknowledged to be part of
```

File: docs/releasing.md
Comment (kashifest) [inline docs/releasing.md:80]:
```suggestion
  the release, but not overwhelming the important changes contained by the release.
```

File: docs/releasing.md
Comment (kashifest) [inline docs/releasing.md:123]:
```suggestion
- Make sure Ironic and Mariadb version references are in place with the correct
```

File: docs/releasing.md
Comment (kashifest) [inline docs/releasing.md:92]:
```suggestion
backwards compatibility, for each CAPM3 release we also create tags in these
```

File: docs/releasing.md
Comment (kashifest) [inline docs/releasing.md:125]:
```suggestion
- If the release you're making isn't the latest release (new
```

File: docs/releasing.md
Comment (kashifest) [inline docs/releasing.md:128]:
```suggestion
- If it is a release candidate(rc) or pre-release , tick pre-release box.
```

File: docs/releasing.md
Comment (kashifest) [inline docs/releasing.md:155]: This should probably point to a file upstream when the PR is merged.

**Review by kashifest — CHANGES_REQUESTED** [review]

File: docs/releasing.md
Comment (furkatgofurov7) [inline docs/releasing.md:88]: Not necessarily before capm3 release I would say, but once CAPM3 is published (at least in the past we used to do so)

File: docs/releasing.md
Comment (furkatgofurov7) [inline docs/releasing.md:116]:
```suggestion
[Mariadb](https://quay.io/repository/metal3-io/mariadb) container images will be built
```

File: docs/releasing.md
Comment (furkatgofurov7) [inline docs/releasing.md:90]:
```suggestion
[Mariadb-image](https://github.com/metal3-io/mariadb-image) repositories.
```

File: docs/releasing.md
Comment (furkatgofurov7) [inline docs/releasing.md:103]: I do not have an exact suggestion, but could we reuse this block later instead of repeating it and somehow say: `follow the exact same steps as above` ?

File: docs/releasing.md
Comment (furkatgofurov7) [inline docs/releasing.md:126]: I think you meant: 

```suggestion
- If the release you're making isn't the actually latest release (new patch release in the latest-1 minor release series), uncheck the
```

File: docs/releasing.md
Comment (furkatgofurov7) [inline docs/releasing.md:147]: why only IPAM and not CAPM3?

File: docs/releasing.md
Comment (furkatgofurov7) [inline docs/releasing.md:162]:
```suggestion
and through the `metal3-dev` group mailing list.
```

**Review by furkatgofurov7 — COMMENTED** [review]
Thanks for the update PR. 
Few comments/suggestions from my side:

File: docs/releasing.md
Comment (furkatgofurov7) [reply docs/releasing.md:155]: or the commit hash

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:37]: Fixed.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:10]: Fixed.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:25]: Fixed.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:27]: Fixed.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:39]: Fixed.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:51]: Replaced with primary release tag, so its not confused with actual main branch.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:54]: Fixed.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:55]: Fixed.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:68]: Fixed.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:80]: Fixed.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:123]: Fixed.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:92]: Fixed.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:125]: Fixed.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:128]: Fixed.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:155]: Yes, it'll point to real document link when it gets merged. For now, it points to PR.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:25]: I clarified it is a GitHub team in the metal-3io organization. I don't have access to the organization page where its defined, nor will 99% of the readers. Admins will know where to look.

Fixed the wording.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:36]: Fixed

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:42]: Fixed.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:45]: CLI is more exact, less prone to change, and easier to document. Rights required are exactly the same.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:90]: Fixed

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:103]: I had it like that before, I'll change it back.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:126]: I don't get the point about `latest-1` ? 

Fixed the wording otherwise.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:147]: Fixed.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:162]: Fixed.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:54]: Fixed.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:55]: Fixed.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:63]: This changes the intended meaning of the sentence. Subdirectories are discussed in the next sentence. 

Fixed the wording othewise.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:88]: Why would you want to do them later? When you go public with a release, why'd you intentionally want to have it half-ready? If you face issues tagging the images, or Quay wouldn't build them etc, then your release is broken.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:116]: Fixed.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:60]: Fixed.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:32]: That is a good point. I added it back.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:79]: Fixed

**Comment by tuminoid** [comment]
Thanks @furkatgofurov7 @kashifest for spellcheck and catching couple missing things. Most of the things I've fixed, but some discussion left open. Let me know what you think.

/hold
There is also dependency to testing metrix PR, so I'll hold this so we get it merged first and can properly link to it.

File: docs/releasing.md
Comment (furkatgofurov7) [reply docs/releasing.md]: The confusing part to me here is: if you release any major/minor release, that would always be the latest release and you do not need to uncheck the "latest release" button. 
But, if it is the patch release from the stable branch which is N-1 where N is latest stable branch (i.e you are publishing 1.3.5 patch release from release-1.3 but latest stable branch you have is release-1.4) then you need to uncheck it

File: docs/releasing.md
Comment (furkatgofurov7) [reply docs/releasing.md:91]: I would not recommend tagging it prior CAPM3 tag pushing, since the tag that is used as a prefix for these are the same as CAPM3 tag you will be pushing and you may end up in a situation where users might get confused with these (ironic tags available but not CAPM3, due to some technical reason when you could not push tag in CAPM3). So, IMO it is better to carry out these steps post CAPM3 tag pushing so we are backed up and referencing something that already exists upstream

**Comment by furkatgofurov7** [comment]
> Thanks @furkatgofurov7 @kashifest for spellcheck and catching couple missing things. Most of the things I've fixed, but some discussion left open. Let me know what you think.

Thank you, one more point I would like to bring up and forgot yesterday:
Can we left some notes in the doc somewhere (whereever you feel it fits, probably after pushing the tags sections) about the need we have to make sure that after tag is pushed, the image in quay has started actually building and the image is built successfully and it is available. 
The reason is, we have had a lot of technical issues in Quay when it comes to image building trigger (i.e trigger disabled, build unsuccessful) so person publishing a release has to make sure that quay has processed the tag and published the corresponding container images properly.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md]: There is only one latest release per repo. In GitHub it means the release that is highlighted in the repository main page. 
The only time you tick the latest box is when you release from top off the latest release branch, otherwise must be unticked. Release from 1.2 branch will never be latest if you have 1.3 releases.

I'll see if there is better wording to make this clearer.

**Comment by tuminoid** [comment]
> > Thanks @furkatgofurov7 @kashifest for spellcheck and catching couple missing things. Most of the things I've fixed, but some discussion left open. Let me know what you think.
> 
> Thank you, one more point I would like to bring up and forgot yesterday: Can we leave some notes in the doc somewhere (whereever you feel it fits, probably after pushing the tags sections) about the need we have to make sure that after tag is pushed, the image in quay has started actually building and the image is built successfully and it is available. The reason is, we have had a lot of technical issues in Quay when it comes to image building trigger (i.e trigger disabled, build unsuccessful) so person publishing a release has to make sure that quay has processed the tag and published the corresponding container images properly.
> 
> Edit: And ^ has to be followed for all container image building (CAPM3/IPAM/BMO/Ironic/mariadb)

I added the release artifacts and mention of successfully built container images there. BUT, as mentioned, we need to agree where to put all that stuff. CAPM3 != Metal3, and hence CAPM3 release process shouldn't contain Metal3 release process etc.

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md:91]: I'd still argue that before you press the publish button, nobody needs to care (won't even notice really) is more correct approach than publish first, and then scramble. Publishing a release sends out emails to everyone watching repo, might even trigger an automated workflow somewhere, so it is much bigger failure do it that way.

**Review by kashifest — APPROVED** [review]

**Comment by tuminoid** [comment]
There is two TODO left, one waiting for testing matrix PR to merge, and another for settling the proper location on the non-CAPM3 stuff. The latter needs input, so please share yours @furkatgofurov7 @kashifest .

**Comment by furkatgofurov7** [comment]
> BUT, as mentioned, we need to agree where to put all that stuff. CAPM3 != Metal3, and hence CAPM3 release process shouldn't contain Metal3 release process etc.

Partially agree, true but at the moment it is the repo, which owns kind of centralized status, and dependant on other repos (IPAM, BMO needs to be uplifted in it is go mods) and also main for other repos (whenever CAPM3 is release, you have to have a new ironic-image tag). 

> for settling the proper location on the non-CAPM3 stuff.

probably a new folder (i.e `release` in metal3-docs repo) ..

**Comment by tuminoid** [comment]
> probably a new folder (i.e `release` in metal3-docs repo) ..

I've a WIP going https://github.com/metal3-io/metal3-docs/pull/321 for adding `releasing.md` under `processes` if you had missed it. It is very bare bones right now, but it seems we agree that all the Metal3 releasing, including dev-env modifications and Ironic/Mariadb tagging should be put there, until BMO has proper release process?

File: docs/releasing.md
Comment (furkatgofurov7) [inline docs/releasing.md:144]: This is still confusing to me, though we discussed it earlier iterations, and also as you pointed out, when:

`you release from top off the latest release branch, otherwise must be unticked. Release from 1.2 branch will never be latest if you have 1.3 releases.`

But above, you are mention both new minor and major which should be ticked as latest when you publish one

**Review by furkatgofurov7 — COMMENTED** [review]
/lgtm cancel

One last comment that I think has to be addressed before I will add it back:

File: docs/releasing.md
Comment (tuminoid) [reply docs/releasing.md]: Fixed, better now?

Comment (furkatgofurov7) [reply docs/releasing.md]: Thanks, it looks a lot better

Comment (furkatgofurov7) [reply docs/releasing.md]: Feel free to resolve it

**Pull request merged**
